Tasks/AzureFunctionAppV1/deploymentProvider/ConsumptionWebAppDeploymentProvider.ts

```diff
@@ -30,6 +38,71 @@ export class ConsumptionWebAppDeploymentProvider extends AzureRmWebAppDeployment
         }
         return storageData;
     }
+
+    private async uploadPackage(storageDetails, deployPackage) : Promise<string> {
+        var defer = Q.defer<string>();
+        let storageAccount = storageDetails["AccountName"];
+        let storageKey = storageDetails["AccountKey"];
+        const blobService = azureStorage.createBlobService(storageAccount, storageKey);
+
+        const containerName = 'azure-pipelines-deploy';
+        const blobName = `package_${Date.now()}.zip`;
+        let fileName;
+
+        switch(deployPackage.getPackageType()){
+            case PackageType.folder:
+                let tempPackagePath = webCommonUtility.generateTemporaryFolderOrZipPath(tl.getVariable('AGENT.TEMPDIRECTORY'), false);
+                let archivedWebPackage = await zipUtility.archiveFolder(deployPackage.getPath(), "", tempPackagePath);
+                tl.debug("Compressed folder into zip " +  archivedWebPackage);
+                fileName = archivedWebPackage;
+            break;
+            case PackageType.zip:
+                fileName = deployPackage.getPath();
+            break;
+            default:
+                throw new Error(tl.loc('Invalidwebapppackageorfolderpathprovided', deployPackage.getPath()));
+        }
+
+        blobService.createContainerIfNotExists(containerName, error => {
+            if (error){
+                defer.reject(error);
+            }
+
+            //upoading package
+            blobService.createBlockBlobFromLocalFile(containerName, blobName, fileName, (error, result) => {
+                if (error) {
+                    defer.reject(error);
+                }
+
+                //generating SAS URL
+                let startDate = new Date();
+                let expiryDate = new Date(startDate);
+                expiryDate.setFullYear(startDate.getUTCFullYear() + 1);
+                startDate.setMinutes(startDate.getUTCMinutes() - 5);
+            
+                let sharedAccessPolicy = {
+                    AccessPolicy: {
+                        Permissions: azureStorage.BlobUtilities.SharedAccessPermissions.READ,
+                        Start: startDate,
+                        Expiry: expiryDate
+                    }
+                };
+            
+                let token = blobService.generateSharedAccessSignature(containerName, blobName, sharedAccessPolicy);
+                let sasUrl = blobService.getUrl(containerName, blobName, token);
+                tl.debug(`SAS URL is: ${sasUrl}`);
+                defer.resolve(sasUrl);
+            });
+        });
+        return defer.promise;
+    }
+
+    private async publishRunFromPackage(sasUrl) {
+        await this.appService.patchApplicationSettings({'WEBSITE_RUN_FROM_PACKAGE': sasUrl});
```

Comment:
Can we have a console statement here to say that we are updating the app settings?

Reply:
console.log(tl.loc('UpdatedRunFromPackageSettings', sasUrl)
UpdatingRunFromPackageSettings: "Updated 'W_R_F_P Application setting to %s"

Reply:
It is done.